src/main/java/com/commercetools/sync/products/utils/ProductVariantUpdateActionUtils.java

```diff
@@ -308,23 +229,116 @@
+     * // TODO: UNIT TESTS
+     *
+     * @param productKey         the key of the product that the variants belong to. It is used only in the error
+     *                           messages if any.
+     * @param oldProductVariant  the {@link ProductVariant} which should be updated.
+     * @param newProductVariant  the {@link ProductVariantDraft} where we get the new list of attributes.
+     * @param attributesMetaData a map of attribute name -&gt; {@link AttributeMetaData}; which defines attribute
+     *                           information: its name, whether a value is required or not and whether it has the
+     *                           constraint "SameForAll" or not.
+     * @param syncOptions        the sync options wrapper which contains options related to the sync process supplied by
+     *                           the user. For example, custom callbacks to call in case of warnings or errors occurring
+     *                           on the build update action process. And other options (See {@link ProductSyncOptions}
+     *                           for more info).
+     * @return a list that contains all the update actions needed, otherwise an empty list if no update actions are
+     *          needed.
+     */
+    @Nonnull
+    public static List<UpdateAction<Product>> buildProductVariantAttributesUpdateActions(
+        @Nullable final String productKey,
+        @Nonnull final ProductVariant oldProductVariant,
+        @Nonnull final ProductVariantDraft newProductVariant,
+        @Nonnull final Map<String, AttributeMetaData> attributesMetaData,
+        @Nonnull final ProductSyncOptions syncOptions) {
+
+        final Integer oldProductVariantId = oldProductVariant.getId();
+        final List<AttributeDraft> newProductVariantAttributes = newProductVariant.getAttributes();
+        final List<Attribute> oldProductVariantAttributes = oldProductVariant.getAttributes();
+
+        final List<UpdateAction<Product>> updateActions = buildRemoveUpdateActions(oldProductVariantAttributes,
+            newProductVariantAttributes, Attribute::getName, AttributeDraft::getName,
+            attribute -> {
+                try {
+                    return buildUnSetAttribute(oldProductVariantId, attribute.getName(), attributesMetaData);
+                } catch (final BuildUpdateActionException buildUpdateActionException) {
+                    final String errorMessage = format(FAILED_TO_BUILD_ATTRIBUTE_UPDATE_ACTION, attribute.getName(),
+                        newProductVariant.getKey(), productKey, buildUpdateActionException.getMessage());
+                    syncOptions.applyErrorCallback(errorMessage, buildUpdateActionException);
+                    return null;
+                }
+            });
+
+        final Map<String, Attribute> oldAttributesMap =
+            collectionToMap(oldProductVariantAttributes, Attribute::getName);
+
+        emptyIfNull(newProductVariantAttributes).forEach(newAttribute -> {
```

Comment:
yes it can, one can define a product variant like that:

````
ProductVariantDraftBuilder.of()
                                  .attributes((List<AttributeDraft>)null)
                                  .build();
````